crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/LayerMetadata.java

```diff
@@ -16,35 +16,41 @@
 
 package com.google.cloud.tools.crepecake.cache;
 
+import com.google.common.annotations.VisibleForTesting;
+import java.nio.file.attribute.FileTime;
 import java.util.List;
 
-/** Metadata about a layer stored in the cache. This is part of the {@link CacheMetadata}. */
+/**
+ * Metadata about an application layer stored in the cache. This is part of the {@link
+ * CacheMetadata}.
+ */
 class LayerMetadata {
 
-  /** The type of layer. */
-  private final CachedLayerType type;
-
   /** The paths to the source files that the layer was constructed from. */
   private List<String> sourceFiles;
 
-  /** The last time the layer was constructed, or negative if unknown. */
-  private long lastModifiedTime;
+  /** The last time the layer was constructed. */
+  private final FileTime lastModifiedTime;
+
+  LayerMetadata(List<String> sourceFiles, FileTime lastModifiedTime) {
+    if (sourceFiles.size() == 0) {
+      throw new IllegalArgumentException("Source files for application layer cannot be empty");
+    }
 
-  LayerMetadata(CachedLayerType type, List<String> sourceFiles, long lastModifiedTime) {
-    this.type = type;
     this.sourceFiles = sourceFiles;
     this.lastModifiedTime = lastModifiedTime;
   }
 
-  CachedLayerType getType() {
-    return type;
-  }
-
   List<String> getSourceFiles() {
     return sourceFiles;
   }
 
-  public long getLastModifiedTime() {
+  public FileTime getLastModifiedTime() {
     return lastModifiedTime;
   }
+
+  @VisibleForTesting
```

Comment:
I may be losing the thread, but is `VisibleForTesting` right here? This seems to be used in non-test code and not used in test code, but I could be mixing up different setSourceFiles methods.

Reply:
`LayerMetadata` is intended to be an immutable class. Its `setSourceFiles` method is only used in `CacheCheckerTest` line 138. The other `setSourceFiles` method that is not just for testing is on `CacheMetadataLayerPropertiesObjectTemplate` used as the JSON template for `LayerMetadata`.